File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/property/fetchoptions/PropertyAssignmentFetchOptions.js
define([ "stjs", "as/dto/common/fetchoptions/FetchOptions", "as/dto/property/fetchoptions/PropertyTypeFetchOptions", "as/dto/person/fetchoptions/PersonFetchOptions",
		"as/dto/property/fetchoptions/PropertyAssignmentSortOptions", "as/dto/semanticannotation/fetchoptions/SemanticAnnotationFetchOptions" ], function(stjs, FetchOptions) {
	var PropertyAssignmentFetchOptions = function() {
	};
	stjs.extend(PropertyAssignmentFetchOptions, FetchOptions, [ FetchOptions ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.property.fetchoptions.PropertyAssignmentFetchOptions';
		constructor.serialVersionUID = 1;
		prototype.entityType = null;
		prototype.propertyType = null;
		prototype.semanticAnnotations = null;
		prototype.registrator = null;
		prototype.sort = null;

		prototype.withEntityType = function() {
			if (this.entityType == null) {
				var EntityTypeFetchOptions = require("as/dto/entitytype/fetchoptions/EntityTypeFetchOptions");
				this.entityType = new EntityTypeFetchOptions();
			}
			return this.entityType;
		};
		prototype.withEntityTypeUsing = function(fetchOptions) {
			return this.entityType = fetchOptions;
		};
		prototype.hasEntityType = function() {
			return this.entityType != null;
		};
		prototype.withPropertyType = function() {
			if (this.propertyType == null) {
				var PropertyTypeFetchOptions = require("as/dto/property/fetchoptions/PropertyTypeFetchOptions");
				this.propertyType = new PropertyTypeFetchOptions();
			}
			return this.propertyType;
		};
		prototype.withPropertyTypeUsing = function(fetchOptions) {
			return this.propertyType = fetchOptions;
		};
		prototype.hasPropertyType = function() {
			return this.propertyType != null;
		};
		prototype.withSemanticAnnotations = function() {
			if (this.semanticAnnotations == null) {
				var SemanticAnnotationFetchOptions = require("as/dto/semanticannotation/fetchoptions/SemanticAnnotationFetchOptions");
				this.semanticAnnotations = new SemanticAnnotationFetchOptions();
			}
			return this.semanticAnnotations;
		};
		prototype.withSemanticAnnotationsUsing = function(fetchOptions) {
			return this.semanticAnnotations = fetchOptions;
		};
		prototype.hasSemanticAnnotations = function() {
			return this.semanticAnnotations != null;
		};
		prototype.withRegistrator = function() {
			if (this.registrator == null) {
				var PersonFetchOptions = require("as/dto/person/fetchoptions/PersonFetchOptions");
				this.registrator = new PersonFetchOptions();
			}
			return this.registrator;
		};
		prototype.withRegistratorUsing = function(fetchOptions) {
			return this.registrator = fetchOptions;
		};
		prototype.hasRegistrator = function() {
			return this.registrator != null;
		};
		prototype.sortBy = function() {
			if (this.sort == null) {
				var PropertyAssignmentSortOptions = require("as/dto/property/fetchoptions/PropertyAssignmentSortOptions");
				this.sort = new PropertyAssignmentSortOptions();
			}
			return this.sort;
		};
		prototype.getSortBy = function() {
			return this.sort;
		};
	}, {
		entityType : "EntityTypeFetchOptions",
		propertyType : "PropertyTypeFetchOptions",
		semanticAnnotations : "SemanticAnnotationFetchOptions",
		registrator : "PersonFetchOptions",
		sort : "PropertyAssignmentSortOptions"
	});
	return PropertyAssignmentFetchOptions;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/property/fetchoptions/PropertyAssignmentFetchOptions.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.property.fetchoptions;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptionsToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.entitytype.fetchoptions.EntityTypeFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.fetchoptions.PersonFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.PropertyAssignment;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.fetchoptions.PropertyTypeFetchOptions;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;

/*
 * Class automatically generated with DtoGenerator
 */
@JsonObject("as.dto.property.fetchoptions.PropertyAssignmentFetchOptions")
public class PropertyAssignmentFetchOptions extends FetchOptions<PropertyAssignment> implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private EntityTypeFetchOptions entityType;

    @JsonProperty
    private PropertyTypeFetchOptions propertyType;

    @JsonProperty
    private PersonFetchOptions registrator;

    @JsonProperty
    private PropertyAssignmentSortOptions sort;

    // Method automatically generated with DtoGenerator
    public EntityTypeFetchOptions withEntityType()
    {
        if (entityType == null)
        {
            entityType = new EntityTypeFetchOptions();
        }
        return entityType;
    }

    // Method automatically generated with DtoGenerator
    public EntityTypeFetchOptions withEntityTypeUsing(EntityTypeFetchOptions fetchOptions)
    {
        return entityType = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasEntityType()
    {
        return entityType != null;
    }

    // Method automatically generated with DtoGenerator
    public PropertyTypeFetchOptions withPropertyType()
    {
        if (propertyType == null)
        {
            propertyType = new PropertyTypeFetchOptions();
        }
        return propertyType;
    }

    // Method automatically generated with DtoGenerator
    public PropertyTypeFetchOptions withPropertyTypeUsing(PropertyTypeFetchOptions fetchOptions)
    {
        return propertyType = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasPropertyType()
    {
        return propertyType != null;
    }

    // Method automatically generated with DtoGenerator
    public PersonFetchOptions withRegistrator()
    {
        if (registrator == null)
        {
            registrator = new PersonFetchOptions();
        }
        return registrator;
    }

    // Method automatically generated with DtoGenerator
    public PersonFetchOptions withRegistratorUsing(PersonFetchOptions fetchOptions)
    {
        return registrator = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasRegistrator()
    {
        return registrator != null;
    }

    // Method automatically generated with DtoGenerator
    @Override
    public PropertyAssignmentSortOptions sortBy()
    {
        if (sort == null)
        {
            sort = new PropertyAssignmentSortOptions();
        }
        return sort;
    }

    // Method automatically generated with DtoGenerator
    @Override
    public PropertyAssignmentSortOptions getSortBy()
    {
        return sort;
    }
    @Override
    protected FetchOptionsToStringBuilder getFetchOptionsStringBuilder()
    {
        FetchOptionsToStringBuilder f = new FetchOptionsToStringBuilder("PropertyAssignment", this);
        f.addFetchOption("EntityType", entityType);
        f.addFetchOption("PropertyType", propertyType);
        f.addFetchOption("Registrator", registrator);
        return f;
    }

}
